tools/rapids-check-pr-job-dependencies

```diff
@@ -5,21 +5,35 @@
 # This is necessary since the RAPIDS branch protections are configured to require
 # the "pr-builder" job to pass for all PRs. It's implied that that job depends
 # on all other jobs in the workflow.
+#
+# An optional argument can be passed to ignore certain jobs from the check.
+# The argument should be a space-separated list of job names, e.g.:
+#  check-pr-builder-depends.sh "job1 job2 job3"
 set -euo pipefail
 
 export WORKFLOW_FILE=${WORKFLOW_FILE:-".github/workflows/pr.yaml"}
 export PR_BUILDER_JOB_NAME=${PR_BUILDER_JOB_NAME:-"pr-builder"}
+export IGNORED_JOBS=${1:-""}
+
+WORKFLOW_JOBS=$(yq '((.jobs | keys) - [strenv(PR_BUILDER_JOB_NAME)])' "${WORKFLOW_FILE}")
+export WORKFLOW_JOBS
+
+PR_BUILDER_JOB_NEEDS=$(yq '(.jobs.[env(PR_BUILDER_JOB_NAME)].needs)' "${WORKFLOW_FILE}")
+export PR_BUILDER_JOB_NEEDS
+
+IGNORED_JOBS=$(yq -n 'strenv(IGNORED_JOBS) | split(" ")')
```

Comment:
```suggestion
IGNORED_JOBS=$(yq -n 'strenv(IGNORED_JOBS) | trim | split(" ")')
```

Would you consider passing this through `trim` ([`yq` docs](https://mikefarah.gitbook.io/yq/operators/string-operators#trim-strings)), so we don't have to think about how `split()` handles leading or trailing whitespace?

And one other related question... how sensitive is this to having more than one consecutive space between items?

I tried to test that myself like this:

```shell
IGNORED_JOBS="abc    def"
yq -n '.strenv(IGNORED_JOBS) | split(" ")'
```

But got this error I don't understand.

```text
jq: error: strenv/1 is not defined at <top-level>, line 1:
(strenv(IGNORED_JOBS) | split(" "))
jq: 1 compile error
```

(that is probably just my lack of familiarity with `yq`)

Reply:
trim is a good idea. i will test this locally real quick and push a change